Remove SysV init support

rjernst opened the pull request:
With the removal of support for older OSes, we no longer have any
supported systems which use SysV init. This commit removes support for
that legacy init system.

relates #51480

rjernst committed:
Remove SysV init support

With the removal of support for older OSes, we no longer have any
supported systems which use SysV init. This commit removes support for
that legacy init system.

relates #51480

elasticmachine commented:
Pinging @elastic/es-core-infra (:Core/Infra/Packaging)

rjernst committed:
remove dead includes

pugnascotia commented on qa/os/centos-7/build.gradle:
```diff
@@ -1 +0,0 @@
-project.ext.shouldTestDocker = true
```
Why has this been removed?

rjernst replied:
It was unintentional, I've added it back

pugnascotia reviewed (dismissed):
Looks fine in general, just one question.

rjernst committed:
Merge branch 'master' into distro_tests40

rjernst committed:
Merge branch 'master' into distro_tests40

rjernst committed:
spotless

rjernst committed:
Do not daemonize when testing error output in packaging tests

The packaging tests start elasticsearch in various ways. All of these
currently expect it is started asynchronously, yet some tests expect it
will fail to start and want to check the error output. This commit adds
a daemonize flag to the utility methods to start elasticsearch for such
cases, so that when the start method returns, all the error output
should already be available since the process will have exited.

relates #51716

rjernst committed:
checkstyle

rjernst committed:
clear windows log file after test

rjernst committed:
spotless

rjernst committed:
Merge branch 'distro_tests45' into distro_tests40

rjernst committed:
remove old use of update-rc

rjernst committed:
Merge branch 'master' into distro_tests40

rjernst committed:
Merge branch 'master' into distro_tests40

rjernst committed:
remove leftover sysinit handling

rjernst committed:
checkstyle

rjernst committed:
add back removed file

rjernst commented:
@pugnascotia I've brought this up to date, can you take another look?

rjernst committed:
remove chkconfig use

rjernst committed:
guard...
